test/end-to-end/company.js

```diff
@@ -1933,4 +1933,195 @@ describe("Company endpoint", () => {
+            password: "password123",
+        };
+
+        const test_random_user = {
+            email: "random@email.com",
+            password: "password123",
+        };
+
+        const test_agent = agent();
+
+        beforeAll(async () => {
+            await test_agent
+                .delete("/auth/login")
+                .expect(HTTPStatus.OK);
+
+            await Company.deleteMany({});
+
+
+            test_company = await Company.create({
+                name: "Cool Company",
+                contacts: ["112", "122"],
+                bio: "Cool company bio",
+                logo: "http://awebsite.com/alogo.jpg",
+                hasFinishedRegistration: true
+            });
+
+            test_random = await Company.create({
+                name: "Random Company",
+                contacts: ["112", "122"],
+                bio: "Random company bio",
+                logo: "http://awebsite.com/alogo.jpg",
+                hasFinishedRegistration: true
+            });
+
+            await Account.deleteMany({});
+
+            await Account.create({
+                email: test_user_admin.email,
+                password: await hash(test_user_admin.password),
+                isAdmin: true
+            });
+
+            await Account.create({
+                email: test_user_company.email,
+                password: await hash(test_user_company.password),
+                company: test_company._id
+            });
+
+            await Account.create({
+                email: test_random_user.email,
+                password: await hash(test_random_user.password),
+                company: test_random._id
+            });
+        });
+
+        afterEach(async () => {
+            await test_agent
+                .delete("/auth/login")
+                .expect(HTTPStatus.OK);
+        });
+
+        afterAll(async () => {
+            await Company.deleteMany({});
+            await Account.deleteMany({});
+        });
+
+        describe("ID Validation", () => {
+            test("Should fail if id is not a valid ObjectID", async () => {
+                const res = await test_agent
+                    .put("/company/123/edit")
+                    .send()
+                    .expect(HTTPStatus.UNAUTHORIZED);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.INSUFFICIENT_PERMISSIONS);
+            });
+
+            test("Should fail if id is not a valid company", async () => {
+                const id = "111111111111111111111111";
+                const res = await test_agent
+                    .put(`/company/${id}/edit`)
+                    .send()
+                    .expect(HTTPStatus.UNAUTHORIZED);
+
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.INSUFFICIENT_PERMISSIONS);
+            });
+        });
+
+        describe("Should fail if different user/company", () => {
+            test("Should fail if different user", async () => {
```

Comment:
Writing a describe with only one test with the same descriptions seems, redundant, I'd remove describe in cases like this one. Alternatively, wrap this and the tests below into one describe

Reply:
Fixed this!